src/components/TrackRow.vue

```diff
@@ -1,24 +1,22 @@
 <template>
+  <div v-if="index !== undefined" class="flex w-8 justify-end">
+    <span>{{ index + 1 }}</span>
+  </div>
   <div class="title flex-2 flex overflow-hidden items-center max-h-12">
     <!-- <a class="disabled" ng-if="song.disabled" ng-click="copyrightNotice()"> song.title </a> -->
     <vue-feather
-      v-if="!isRedHeart(song.id)"
-      class="cursor-pointer flex-none"
+      class="cursor-pointer flex-none mx-4"
       type="heart"
       size="18"
       stroke-width="1"
-      stroke="#666666"
-      @click="setRedHeart(toRaw(song), true)" />
-    <vue-feather
-      v-if="isRedHeart(song.id)"
-      class="cursor-pointer flex-none"
-      type="heart"
-      fill="red"
-      stroke="red"
-      size="18"
-      @click="setRedHeart(toRaw(song), false)" />
+      :stroke="isRedHeart(song.id) ? 'red' : '#666666'"
+      :fill="isRedHeart(song.id) ? 'red' : 'none'"
+      @click="setRedHeart(toRaw(song), !isRedHeart(song.id))" />
 
-    <a class="cursor-pointer ml-3 truncate min-w-0" @click="play(song)">{{ song.title }}</a>
+    <a class="cursor-pointer truncate min-w-0" @click="play(song)">{{ song.title }}</a>
+    <span v-if="showSource" class="px-1 mx-4 text-gray-500 text-ms text-center border border-solid rounded border-gray-500">
+      {{ song.source }}
```

Comment:
```suggestion
      {{ t(song.source) }}
```
Add translated source name